clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -172,6 +173,115 @@ public void testOverwriteAcksAndRetriesForIdempotentProducers() {
                 config.getString(ProducerConfig.TRANSACTIONAL_ID_CONFIG)));
     }
 
+    @Test
+    public void testAcksAndIdempotenceForIdempotentProducers() {
+        Properties validProps = new Properties() {{
+            setProperty(
+                ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+            setProperty(
+                ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+            setProperty(
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
```

Comment:
We could move the common configs to a base properties instance and invoke putAll() in each of these instead of repeating these 3 configs for each case.